### PR TITLE
Exorelay-js: retry when connection closed

### PR DESCRIPTION
**Pull request by hugobho**
Retry connection to ExoCom when websocket is closed.

### CODE REVIEW

File: javascript/exorelay/features/support/steps-given.ls [hunk 1]
```diff
@@ -19,6 +20,12 @@ module.exports = ->
     @exo-relay = new ExoRelay {exocom-host: 'localhost', @exocom-port, role: @role = 'test-service'}
 
 
+  @Given /^an ExoRelay instance is connected at port (\d+)$/ (@exocom-port, done) ->
+    @exo-relay = new ExoRelay {exocom-host: 'localhost', @exocom-port, role: @role = 'test-service'}
+      ..connect!
+      ..on 'online', ~> done!
```
Comment (hugobho) [inline javascript/exorelay/features/support/steps-given.ls]: This looks weird, but apparently ExoRelay makes a callback with the Exocom port as the first parameter of an `'online'` event. We're not using it so I figured I'd ignore it. We should probably make a separate PR that changes that callback to `(err, port)`

Comment (charlierudolph) [inline javascript/exorelay/features/support/steps-given.ls]: I've never seen event emitters follow `(err, result)` as they typically have a dedicated `error` event. So this is fine

File: javascript/exorelay/src/websocket-connector/websocket-connector.ls [hunk 2]
```diff
@@ -60,6 +61,12 @@ class WebSocketConnector extends EventEmitter
       ..on 'open', @_on-socket-open
       ..on 'message', @_on-socket-message
       ..on 'error', @_on-socket-error
+      ..on 'close', @_on-socket-close
+
+
+  _on-socket-close: ~>
+    unless @socket-killed
+      @connect
```
Comment (charlierudolph) [inline javascript/exorelay/src/websocket-connector/websocket-connector.ls]: shouldn't you be calling connect? Please investigate why the test is passing before fixing this.

File: javascript/exorelay/src/websocket-connector/websocket-connector.ls [hunk 3]
```diff
@@ -27,6 +27,7 @@ class WebSocketConnector extends EventEmitter
   close: ->
     return unless @socket
     debug "no longer connected at 'ws://#{@exocom-host}/#{@exocom-port}'"
+    @socket-killed = yes
```
Comment (charlierudolph) [inline javascript/exorelay/src/websocket-connector/websocket-connector.ls]: Should we reset `socket-killed` to no somewhere? Otherwise after closing it, if you were to connect again it would not have the reconnect behavior.

**Comment by charlierudolph** [comment]
@hugobho any updates here? I can help out with this PR as well if you'd like

**Comment by charlierudolph** [comment]
closing in favor of #158
